src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs

```diff
@@ -107,14 +108,35 @@ public AuthenticationContext(string authority, bool validateAuthority, TokenCach
         {
         }
 
+        /// <summary>
+        /// Constructor to create the context with the address of the authority and flag to turn address validation off.
+        /// Using this constructor, address validation can be turned off. Make sure you are aware of the security implication of not validating the address.
+        /// 
+        /// This constructor allows the HttpClient to be provided instead of having a new one created.  This can be useful when working with multiple different tenants.
+        /// </summary>
+        /// <param name="authority">Address of the authority to issue token.</param>
+        /// <param name="validateAuthority">Flag to turn address validation ON or OFF.</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken</param>
+        public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache, HttpClient httpClient)
```

Comment:
While I am not opposed to the implementation on functional grounds, adding a new constructor is frawned upon in ADAL because users already find the API hard to use. Still, I understand that not being able to specify a custom HttpClient is preventing some Xamarin users from using ADAL. 

@jmprieur @henrik-me - please advise.

Reply:
Overall, given ADAL doesn't have the ConfigObject, I think this is acceptable, though I'm not sure it's enough for all scenarios we would need to support.

---
In reply to: [246794805](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1472#discussion_r246794805) [](ancestors = 246794805)